### DISCUSSION TITLE
unable to run any sdxl models with AMD rx6600 on windows

### DISCUSSION THREAD

**Discussion by patientx** [vladmandic/sdnext#2380] [Q&A]
Python 3.10.11 on Windows
Version: app=sd.next updated=2023-10-22 hash=0288106a 
PC : ryzen 3600x , 16 gb ddr4-3200, amd rx 6600 8 gb gpu, win10 

command line parameters : webui.bat --backend diffusers --medvram or just  webui.bat --backend diffusers

I tried the default model than later downloaded this one : copaxTimelessxlSDXL1_v7 . 
Started the app with diffusers backend and medvram also tried without the second option, also the "Enable model CPU offload (--medvram)" is selected in diffusers menu in options. 

In the end this is the resulting log 

`03:03:15-429615 STATUS   Select: model="copaxTimelessxlSDXL1_v7 [55818ae18a]"
Loading weights: D:\stable-diffusion-webui-directml\models\Stable-diffusion\copaxTimelessxlSDXL1_v7.safetensors ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.0/6.9 GB -:--:--
03:03:15-574647 STATUS   Torch override dtype: no-half set
03:03:15-576649 STATUS   Torch override VAE dtype: no-half set
03:03:15-577648 STATUS   Setting Torch parameters: device=cpu dtype=torch.float32 vae=torch.float32 unet=torch.float32 context=no_grad fp16=False bf16=False
03:03:15-579650 STATUS   Autodetect: model="Stable Diffusion XL" class=StableDiffusionXLPipeline file="D:\stable-diffusion-webui-directml\models\Stable-diffusion\copaxTimelessxlSDXL1_v7.safetensors" size=6617MB
03:03:43-320962 STATUS   Loaded embeddings: loaded=0 skipped=0 time=0.02s
03:03:50-155493 STATUS   Loaded model: time=28.01s { load=28.01s } native=1024 {'ram': {'used': 6.47, 'total': 15.93}}
03:03:50-412550 STATUS   Startup time: 54.41s { torch=8.05s gradio=0.88s diffusers=0.06s libraries=1.86s extensions=4.75s face-restore=0.31s upscalers=0.13s ui-extra-networks=0.47s ui-img2img=0.07s ui-settings=0.17s ui-extensions=1.52s
                         ui-defaults=0.07s launch=0.26s api=0.09s app-started=0.63s checkpoint=34.97s }
03:03:55-626723 STATUS   Browser session: client=127.0.0.1 agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36
03:04:05-649971 ERROR    Prompt parser encode: Torch not compiled with CUDA enabled
03:04:05-652971 STATUS   Torch not compiled with CUDA enabled
03:04:05-654971 ERROR    Exception: local variable 'output' referenced before assignment
03:04:05-655972 ERROR    Arguments: args=('task(ej6qsipai0q4dzn)', 'a bowl of oranges, oil painting\n', 'noisy, blurry, deformed, (deformed, distorted, disfigured:1.3),poorly drawn,bad anatomy,wrong anatomy,extra limb,missing
                         limb,floating limbs,(mutated hands and fingers:1.4),disconnected big limbs, mutation, mutated, ugly, disgusting, blurry, amputation, big face, big head,text,logo\n', [], 20, None, None, True, True, False, 1, 1,
                         7, 6, 0.7, 1, 3530596996.0, -1.0, 0, 0, 0, 1024, 1024, False, 0.5, 2, 'None', False, 20, 0, 0, 5, 0.8, '', '', [], 0, False, False, 'positive', 'comma', 0, False, False, '', 0, '', [], 0, '', [], 0, '', [],
                         False, True, False, False, False, False, 0, 0, 2, 512, 512, True, 'None', 'None', 0, 0, 0.2, None, False, '0', '0', 'inswapper_128.onnx', 'CodeFormer', 1, True, 'ESRGAN TGHQFace8x_500k', 2, 1, False, True, 0, 0,
                         0, False, 0.5) kwargs={}
03:04:05-660973 ERROR    gradio call: UnboundLocalError
╭───────────────────────────────────────────────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────────────────────────────────────────────╮
│ D:\automatic\modules\call_queue.py:34 in f                                                                                                                                                           │
│                                                                                                                                                                                                      │
│   33 │   │   │   try:                                                                                                                                                                                │
│ ❱ 34 │   │   │   │   res = func(*args, **kwargs)                                                                                                                                                     │
│   35 │   │   │   │   progress.record_results(id_task, res)                                                                                                                                           │
│                                                                                                                                                                                                      │
│ D:\automatic\modules\txt2img.py:66 in txt2img                                                                                                                                                        │
│                                                                                                                                                                                                      │
│   65 │   if processed is None:                                                                                                                                                                       │
│ ❱ 66 │   │   processed = processing.process_images(p)                                                                                                                                                │
│   67 │   p.close()                                                                                                                                                                                   │
│                                                                                                                                                                                                      │
│ D:\automatic\modules\processing.py:687 in process_images                                                                                                                                             │
│                                                                                                                                                                                                      │
│    686 │   │   │   with context_hypertile_vae(p), context_hypertile_unet(p):                                                                                                                         │
│ ❱  687 │   │   │   │   res = process_images_inner(p)                                                                                                                                                 │
│    688 │   finally:                                                                                                                                                                                  │
│                                                                                                                                                                                                      │
│ D:\automatic\modules\processing.py:844 in process_images_inner                                                                                                                                       │
│                                                                                                                                                                                                      │
│    843 │   │   │   │   from modules.processing_diffusers import process_diffusers                                                                                                                    │
│ ❱  844 │   │   │   │   x_samples_ddim = process_diffusers(p, p.seeds, p.prompts, p.negative_pro                                                                                                      │
│    845 │   │   │   else:                                                                                                                                                                             │
│                                                                                                                                                                                                      │
│ D:\automatic\modules\processing_diffusers.py:497 in process_diffusers                                                                                                                                │
│                                                                                                                                                                                                      │
│   496 │   if not is_refiner_enabled:                                                                                                                                                                 │
│ ❱ 497 │   │   results = vae_decode(latents=output.images, model=shared.sd_model, full_quality=                                                                                                       │
│   498                                                                                                                                                                                                │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
UnboundLocalError: local variable 'output' referenced before assignment`

**Comment by vladmandic** [comment]
typically people open issues when they have a simple question, but this is clearly an issue - can you create it as such and fill the template?
also, try to quote the log a bit nicer, this is not really readable.

**Comment by vladmandic** [comment]
btw, you do not have torch-directml installed, you're attempting to run with cpu only.

> 03:03:15-577648 STATUS   Setting Torch parameters: device=cpu dtype=torch.float32 vae=torch.float32 unet=torch.float32 context=no_grad fp16=False bf16=False 

i would run `webui --use-directml --reinstall` to force reinstalling of torch-directml and look at the actual output of install if it installs correctly or not. could be for example that you're using python 3.11 or 3.12 and there is no torch-directml for that python, only python 3.10 (that is a torch-directml thing, not my choice).

**Reply by patientx** [reply]
python is 3.10.11
nope, directml is installed and working correctly when I use --use-directml --medvram to start and generate with sd 1.5 models. I've been using the app this way for months, just after reading on reddit people suggesting that sdxl would run on amd gpu -which I've already tried with sdnext before- with sdnext I've decided to try again. So I've noticed the torch cpu only thing but thought that might be how diffusers option worked , just now tried with a bunch of sd 1.5 models all generated correctly.
